src/App.tsx

```diff
@@ -14,11 +16,125 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const numbers: number[] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const App: React.FC = () => {
+  const selectedGoods = [...goodsFromServer];
+  const numbersSelect = [...numbers];
+
+  const [isVisible, setIsVisible] = useState(false);
+  const [numberValue, setNumberValue] = useState(0);
+  const [sortBy, setSortBy] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reverse = () => {
+    setIsReversed(current => !current);
+  };
+
+  const sortAb = () => {
+    setSortBy('alphab');
+  };
+
+  const sortLength = () => {
+    setSortBy('length');
+  };
+
+  const visibleGoods = selectedGoods.filter(good => (
+    good.length >= numberValue
+  ));
+
+  visibleGoods.sort((g1, g2) => {
+    switch (sortBy) {
+      case 'alphab':
+        return g1.localeCompare(g2);
+
+      case 'length':
+        return g1.length - g2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
```

Comment:
it is better to create functions for array sorting and reversing and call functions in onClick